### PR TITLE
Fix pinball BCs

### PR DESCRIPTION
**Pull request by jcallaham**
There was a bug in the derivation of boundary conditions for flows with multiple actuators.  Basically the code initially set the "external" BCs to zero and then set each "control" BC individually to get velocity fields with the right values.  But it didn't reset the external BCs to zero in between, so the inflow was incorrectly nonzero for the BC velocity fields.  This happened to cancel out if you had exactly asymmetric actuation.

Also found an easier way of deriving those BC fields in a more natural way for Firedrake (via projection, instead of asking for the Riesz representation of an assembled weak form).

### CODE REVIEW

**Comment by jcallaham** [comment]
@ludgerpaehler any idea why the CI test for black is failing on "/github/workspace/docs/conf.py"?

File: examples/pinball/run-transient.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import psutil
+
+import hydrogym.firedrake as hgym
+
+output_dir = "."
+pvd_out = None
+restart = None
+checkpoint = "checkpoint-coarse.h5"
+
+flow = hgym.Pinball(Re=10, h5_file=restart, mesh="coarse")
```
Comment (nzolman) [inline examples/pinball/run-transient.py:10]: Are we still using the `hf_file` kwarg? I thought I had seen that checkpoints are loaded with the `restart` keyword. https://github.com/dynamicslab/hydrogym/blob/main/hydrogym/core.py#L54

**Comment by ludgerpaehler** [comment]
Just finishing off KoopmanRL, will begin to review later s.t. we get everything aligned

**Comment by jcallaham** [comment]
> @ludgerpaehler any idea why the CI test for black is failing on "/github/workspace/docs/conf.py"?

Never mind - this got missed in my formatting calls for some reason.  Should be okay now.

File: examples/pinball/run-transient.py (hunk 1 above)
Comment (jcallaham) [reply examples/pinball/run-transient.py:10]: Thanks, good catch.  I copy-pasted from the demo script, which is also out of date.

**Comment by jcallaham** [comment]
Closing - code was merged in #134

**Pull request closed without merging**